_posts/2023-09-11-Rails-7-1-0-beta-1-has-been-released.markdown

```diff
@@ -0,0 +1,143 @@
+---
+layout: post
+title: 'Rails 7.1.0 Beta 1: Dockerfiles, Attribute Normalization, Single Use Tokens and More'
+categories: releases
+author: rafaelfranca
+published: true
+date: 2023-09-11 21:00:00 -04:00
+---
+
+RailsWorld is going to be soon, and to prepare for the celebration of the 20th anniversary of Rails, we are releasing the first beta of Rails 7.1.0!
+
+There are [over five thousand commits](https://github.com/rails/rails/compare/v7.0.8...main) since Rails 7.0, so this release is packed with new features and improvements.
+
+Please help us test all this new stuff so we can ensure a solid final release of Rails 7.1.
+
+## Generate Dockerfile for new applications
+
+Rails will now generate a Docker-related files when you run `rails new`.
+
+These files serve as a foundational setup for deploying your Rails application in a
+production environment using Docker.
+
+## Building blocks for authentication
+
+To complement with `has_secure_password` Rails 7.1 brings a few new building blocks to help developers to build their
+own authentication system.
+
+First, [`normalizes`](https://edgeapi.rubyonrails.org/classes/ActiveRecord/Normalization/ClassMethods.html#method-i-normalizes)
+declares an attribute normalization. This is useful to normalize attributes before saving them to the database.
+
+```ruby
+class User < ApplicationRecord
+  normalizes :email, with: -> email { email.strip.downcase }
+end
+
+user = User.create(email: " CRUISE-CONTROL@EXAMPLE.COM\n")
```

Comment:
```suggestion
user = User.create(email: "CRUISE-CONTROL@EXAMPLE.COM\n")
```

Reply:
Space if by design. We are showing the strip call

Reply:
Aaaaaa I missed that! 👍🏾